kore/src/Kore/Equation/Application.hs

```diff
@@ -222,14 +222,14 @@ attemptEquation sideCondition termLike equation =
     simplifyArgumentWithResult
         argument
         antiLeft
-        (matchPredicate, matchSubstitution)
```

Comment:
Will this drop the `matchPredicate`? I don't see where it's included, otherwise.

Reply:
Yes, I can just make the function take only the substitution if it's clearer.

Reply:
We can't drop the `matchPredicate`, though. We just want to postpone checking it until the `checkRequires` stage.

Reply:
Huh, for some reason I was convinced we already use it in `checkRequires`. You're right, thank you for noticing this!